sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -298,7 +344,9 @@ CURLcode CurlSession::SendBuffer(uint8_t const* buffer, size_t bufferSize)
       }
     };
   }
-
+#ifdef WINDOWS
+  WinSocketSetBuffSize(this->m_curlSocket, this->m_logger);
+#endif // WINDOWS
```

Comment:
Same here. Can we move this to before the loop itself. 

Also, would it be possible to move it into the caller, outside, and before the start of, the while(...) loop in `UploadBody`?

https://github.com/Azure/azure-sdk-for-cpp/blob/7c3257879870369560358eb35b664f7a75b5d269/sdk/core/azure-core/src/http/curl/curl.cpp#L368-L381